frontend/HelperFuncs/utils.js

```diff
@@ -400,11 +400,81 @@ returns the ID for the most recent transaction associated with [userId]
 */
 export const fetchLatestTransID = async(userId) => {
     const response = await fetch(`http://localhost:3000/transId/${userId}`, 
-    { method: 'GET' })
+        { method: 'GET' })
     if(!response.ok){
         throw new Error('Network response was not ok at fetchLatestTransId', Error);
     }
     // endpoints cannot return ints so we have to convert int -> string -> int
     const data = await response.text();
     return  parseInt(data, 10);
+}
+
+/*
+helper function to return the list of goals associated with the user @ [userId]
+*/
+export const fetchActiveGoals = async(userId) => {
+    const response = await fetch(`http://localhost:3000/activegoals/${userId}`,
+        { method: 'GET' })
+    if(!response.ok){
+        throw new Error('Network response was not ok at fetchActiveGoals', Error);
+    }
+    const data = await response.json();
+    return data;
+}
+
+
+/* 
+analyzes a users transaction data to add to the database of goals by finding high-level generic goals
+that may apply to many users so there are goals to track as soon as a user enters the goal page
+*/
+export const findGenericGoals = async(transactionData) => {
```

Comment:
this is a good start for the logic for this TC, looking forward to the upcoming iterations.